tests/finish-branch-auto-chain.test.sh

```
@@ -0,0 +1,153 @@
+# no backtick/dollar re-evaluation).
+assert_grep "$FINISH_BRANCH" \
+  'TITLE_FILE=\$\(mktemp\)|TITLE_FILE=' \
+  "TITLE_FILE mktemp pattern present"
+assert_grep "$FINISH_BRANCH" \
+  "<<'PR_TITLE_EOF'|<<\"PR_TITLE_EOF\"|<<PR_TITLE_EOF" \
+  "title written via single-quoted heredoc (no shell expansion of agent text)"
+assert_grep "$FINISH_BRANCH" \
+  'IFS= read -r PR_TITLE_VAR' \
+  "title read back into a bash variable for safe quoted expansion"
+assert_grep "$FINISH_BRANCH" \
+  'gh pr create --title "\$PR_TITLE_VAR"' \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)